--- airnav_rtlpower.c.old	2023-02-11 14:50:18.110184014 +0100
+++ airnav_rtlpower.c	2023-02-11 14:50:37.976570947 +0100
@@ -178,10 +178,10 @@
         curl_easy_setopt(curl, CURLOPT_READDATA, (void *) &hd);
 
         /* set the ioctl function */
-        curl_easy_setopt(curl, CURLOPT_IOCTLFUNCTION, rtlpower_my_ioctl);
+        curl_easy_setopt(curl, CURLOPT_SEEKFUNCTION, rtlpower_my_ioctl);
 
         /* pass the file descriptor to the ioctl callback as well */
-        curl_easy_setopt(curl, CURLOPT_IOCTLDATA, (void *) &hd);
+        curl_easy_setopt(curl, CURLOPT_SEEKDATA, (void *) &hd);
 
         /* enable "uploading" (which means PUT when doing HTTP) */
         curl_easy_setopt(curl, CURLOPT_UPLOAD, 1L);
